unit_tests/stencil_composition/sid/test_sid_helpers.cpp

```diff
@@ -0,0 +1,45 @@
+/*
+ * GridTools
+ *
+ * Copyright (c) 2014-2019, ETH Zurich
+ * All rights reserved.
+ *
+ * Please, refer to the LICENSE file in the root directory.
+ * SPDX-License-Identifier: BSD-3-Clause
+ */
+
+#include <gridtools/stencil_composition/sid/sid_helpers.hpp>
+
+#include <gtest/gtest.h>
+
+#include <gridtools/common/defs.hpp>
+#include <gridtools/common/hymap.hpp>
+#include <gridtools/stencil_composition/dim.hpp>
+#include <gridtools/stencil_composition/sid/synthetic.hpp>
+
+namespace gridtools {
+    namespace {
+        using sid::property;
+        using namespace literals;
+
+        TEST(sid_helpers, shifted) {
+            double data[3][5][7];
+
+            using strides_t = hymap::keys<dim::i, dim::j, dim::k>::
+                values<integral_constant<int_t, 5 * 7>, integral_constant<int_t, 7>, integral_constant<int_t, 1>>;
+
+            auto src = sid::synthetic()
+                           .set<property::origin>(sid::host_device::make_simple_ptr_holder(&data[0][0][0]))
+                           .set<property::strides>(strides_t{});
+
+            using offset_t = hymap::keys<dim::i, dim::j>::values<integral_constant<int_t, 1>, int_t>;
+            offset_t offset = {integral_constant<int_t, 1>{}, 2};
```

Comment:
alternatively you can do:
```
auto offset = tuple_util::make<hymap:::keys<dim::i, dim::j>::values>(1c, 2);
```

Reply:
Thanks! I wanted to use this pattern, but couldn't find it.